### PR TITLE
[getting started] Add info on code formatting

### PR DESCRIPTION
**Pull request by ktoso**
This is far too easy to miss or not know about today.

Seems worthwhile to call it out in the getting started guide IMHO. I for one went along for a far too long time without realizing there's a formatter tool we can use.

### CODE REVIEW

**Comment by varungandhi-apple** [comment]
In https://github.com/apple/swift/blob/main/docs/HowToGuides/FirstPullRequest.md, we mention 

> Next, [format your changes](https://github.com/apple/swift/blob/main/docs/HowToGuides/FAQ.md#how-do-i-format-my-changes) using clang-format.

which includes a link on how to format changes. So I'm not sure if moving that to the Getting Started guide is valuable, especially given that the Getting Started guide is already quite long.

That said, I'm definitely biased because I was the one who wrote it all down.

**Comment by gottesmm** [comment]
@varungandhi-apple I think the big picture is that we need to make it more visible. We could have it in both places. It is important to really make it pop out so that people do it. We could also set up tooling to check via a swift-ci test that does git-clang-format on HEAD and expects no change and aborts if we detect a change or something like that.

**Comment by varungandhi-apple** [comment]
I'm okay with linking the text from multiple places, and I'm also okay/weakly against this particular change (we should probably fix the link in the First pull request guide though), but duplicating it in different places doesn't seem like a good idea...

Wrt having a CI setup, I would prefer that any such bot automatically fix the commits before merging instead of yelling at the PR author to go do something it could've done by itself.

**Comment by gottesmm** [comment]
@varungandhi-apple I don't know how we would do that in GitHub @shahmishal do you have any thoughts?

**Comment by varungandhi-apple** [comment]
Akin to swift-ci's test and merge action, perhaps we could forbid merging by default and have a command swift-ci merge. That can create a separate branch with the changes, reformat the changes there, and then rebase changes onto `main`, and delete the temporary branch. It needs to be a little careful about preserving the author/committer metadata etc. I don't know if the GitHub API provides a way to check if the PR was merged though, if the PR gets marked "Closed" instead of "Merged" that would be unfortunate.

**Comment by ktoso** [comment]
We do "fail if nor formatted" in all server side swift repositories and it works well. It's important to leave it to users IMHO because sometimes for matters can do something weird -- perhaps clang format is trustworthy though. 

Swift formatters on the other hand sometimes even did breaking api changing changes which was mind blowing, so we definitely want to leave running formatting to a human before merging.

Okey I'll fix the link and not remove from the other spot but just link to it from the getting started guide as well, seems that would be acceptable?

I've worked on this code base for half a hear and never realized we have a formatted guideline so there's something definitely wrong with the discoverability of this info -- I do often read the getting started guides and contributing guides for hits tho, so adding it here definitely helps imho

**Comment by varungandhi-apple** [comment]
> We do "fail if not formatted" in all server side swift repositories and it works well. It's important to leave it to users IMHO because sometimes for matters can do something weird -- perhaps clang format is trustworthy though.

I don't want to derail this PR with discussions on one-vs-the-other but I think we should prioritize the 99% use case first, which is that the formatter will not mess up the meaning of the code or not format it too shabbily.

> Okey I'll fix the link and not remove from the other spot but just link to it from the getting started guide as well, seems that would be acceptable?

That also seems reasonable IMO. 👍🏼

File: docs/HowToGuides/FAQ.md
Comment (import-brain) [inline docs/HowToGuides/FAQ.md:133]: I personally don't think this should be deleted from FAQ, you could have it in both places.

**Review by ejcheng — CHANGES_REQUESTED** [review]

File: docs/HowToGuides/FAQ.md
Comment (import-brain) [reply docs/HowToGuides/FAQ.md:133]: That way, it could be visible in both docs and help more people out.

**Pull request closed without merging**
